### ISSUE TITLE
Motioneye not accepting more than 2 cameras at once

### ISSUE THREAD

**Issue by rcho5840** [motioneye-project/motioneye#3018]
![image](https://github.com/motioneye-project/motioneye/assets/134464812/29344f30-1a1e-4565-a4ad-3dbec446c3cc)

I am using a raspberry pi 4 with bullseye and 0.42 for my motioneye version. As you can see above, anything above 2 cameras leads to issues with display. The leftmost camera just shows a gray icon with a crossed out camera. I don't believe this is an issue with the cameras, because if I remove a working camera, the grayed out camera will start working. Even when I reduce the cameras to 5 fps, the same issue still occurs.

Does anyone know where this issue is stemming from? I need to have a 5 camera system, so suggestions would also be appreciated.

**Comment by MichaIng** [comment]
First of all, please try it with the motionEye 0.43.1 pre-release.

2nd, please give some more details about the camera types. I guess these are all network cameras?

Generally this reads like a bandwidth and/or processing limitation. Note that for network cameras, every cam causes doubled bandwidth, first for retrieving the stream, motion detection on the host, then sending out out to browser. You can test this by adding the cams as simple MJPEG cameras, which are directly streamed from cam to browser, hence no traffic and processing at the motionEye host.

Aa for processing: check CPU usage. You might run better with a recent Debian/RPi OS Bookworm and more recent FFmpeg, for better hardware acceleration.

**Comment by Marijn0** [comment]
You mentioned in this comment: https://github.com/motioneye-project/motioneye/issues/3025#issue-2389269375

The use of two USB webcams, and I see some V4L devices. It is likely that you are hitting the limitations of the USB controller on the PI 4: https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#raspberry-pi-4

Hence, removing one camera makes the other one work.

Also see the section under Video4Linux devices here:
https://motion-project.github.io/motion_config.html

which states:

> USB cameras take a lot of bandwidth. A USB camera connected to a USB2 port or hub consumes virtually all the bandwidth that the port can handle. Even with a small framesize and low framerate you should not expect to have more than one camera per USB controller. If you need more than one USB camera it is possible to add extra USB PCI cards to your computer. It is also possible that your computer has more than one controller for different USB slots. For example, if the computer has six USB ports, the three on the left may be on controller A versus the three on the right may be on controller B. In this situation, connecting two cameras to ports only on the left would not work due to excessive bandwidth. However connecting one camera to port on the right and the other to the port on the left may work since they are on different controllers.

**Comment by MichaIng** [comment]
Makes sense, also powering could be an issue if those cameras are not powered with a dedicated PSU, since the RPi 4 provides max 1.2A*5V=6W shared across all USB ports.

Kernel and motionEye logs could provide some details:
```sh
dmesg
journalctl -u motioneye
```

Anyway, since the issue is somewhat old and setup/hardware-specific, I am closing it. Feel free to reopen if still active.
